src/Analyzers/VisualBasic/Analyzers/ParenthesesStyle/VisualBasicParenthesesStyleDiagnosticAnalyzer.vb

```diff
@@ -0,0 +1,62 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+
+Imports Microsoft.CodeAnalysis.Diagnostics
+Imports Microsoft.CodeAnalysis.VisualBasic.Precedence
+Imports Microsoft.CodeAnalysis.VisualBasic.LanguageServices
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+Imports Microsoft.CodeAnalysis.RemoveUnnecessaryParentheses
+Imports Microsoft.CodeAnalysis.Precedence
+Imports Microsoft.CodeAnalysis.LanguageServices
+Imports Microsoft.CodeAnalysis.CodeStyle
+Imports System.Collections.Immutable
+Imports Microsoft.CodeAnalysis.Options
+Imports Microsoft.CodeAnalysis.VisualBasic.CodeStyle
+Imports Microsoft.CodeAnalysis.Operations
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.ParenthesesStyle
+    <DiagnosticAnalyzer(LanguageNames.VisualBasic)>
+    Friend Class VisualBasicParenthesesStyleDiagnosticAnalyzer
+        Inherits AbstractBuiltInCodeStyleDiagnosticAnalyzer
+
+        Private Shared ReadOnly s_descriptorForInvocations As New DiagnosticDescriptor(IDEDiagnosticIds.ParenthesesStyleDiagnosticId,
+                                                                              New LocalizableResourceString(NameOf(VisualBasicAnalyzersResources.Add_invocation_parentheses), VisualBasicAnalyzersResources.ResourceManager, GetType(VisualBasicAnalyzersResources)),
+                                                                              New LocalizableResourceString(NameOf(VisualBasicAnalyzersResources.Add_invocation_parentheses), VisualBasicAnalyzersResources.ResourceManager, GetType(VisualBasicAnalyzersResources)),
+                                                                              category:="Not sure what to put here.",
```

Comment:
please look at other diagnostic analyzers.  there shoudl likely be a CodeStyle category or something simialr.

Reply:
I think you should be able to use the static helper method `CreateDescriptor` on the base type and would not need to pass any category, severity, etc.